packages/venia-ui/lib/components/App/localeProvider.js

```diff
@@ -8,15 +8,26 @@ const locale = fromReactIntl(language);
 const LocaleProvider = props => {
     const [messages, setMessages] = useState(null);
 
+    /**
+     * At build time, `__fetchLocaleData__` is injected as a global. Depending on the environment, this global will be
+     * either an ES module with a `default` property, or a plain CJS module.
+     *
+     * Please see {LocalizationPlugin} at @magento/pwa-buildpack/WebpackTools/plugins/LocalizationPlugin.js
+     */
+    const fetchLocale =
+        'default' in __fetchLocaleData__
+            ? __fetchLocaleData__.default
+            : __fetchLocaleData__;
+
     useEffect(() => {
-        import(`../../i18n/${locale}.json`)
+        fetchLocale(locale)
             .then(data => {
                 setMessages(data.default);
             })
             .catch(error => {
                 console.error(`Unable to load translation file. \n${error}`);
             });
-    }, [setMessages]);
+    }, [fetchLocale, setMessages]);
```

Comment:
Do we want this effect to be called when `fetchLocale` changes? It seems like a one-shot operation to me, and should not have any dependency.

Reply:
You're right `fetchLocale` currently won't change, however if we get hot reload support working for translations this could be the trigger which updates the translations.

Also our linter cares about this dependency so rather than forcing the linter not to care I vote we just leave it in as it won't have any negative side effects I believe.